### PR TITLE
Refactor ConsentCard to be a generic FollowCard

### PR DESCRIPTION
**Pull request by rahilb**
## What does this change?

Refactors ConsentCard & ConsentCardList types to be a generic FollowCards, so we may use them for email lists (eventually). 

Introduces a `CardLike` type that the Card(List) accepts: a CardLike has an `id`, a `name`, and a `description`. Introduces a `Followable<T: CardLike>` type that can encapsulate the logic of actually doing a follow... example usage:

```js
                        <FollowCard
                            followable={{
                                value: consent,
                                onChange: newValue => {
                                    this.toggleConsent(newValue, consent.id);
                                },
                            }}
                            hasFollowed={hasConsented}
                        />
```

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/identity/upsell/consent-card/FollowCard.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// @flow
+import React, { Component } from 'preact-compat';
```
Comment (rahilb) [inline static/src/javascripts/projects/common/modules/identity/upsell/consent-card/FollowCard.js]: This file was renamed, and modified quite a bit... GitHub thinks it's a new file.

File: static/src/javascripts/projects/common/modules/identity/upsell/consent-card/FollowCard.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    isOptOut: boolean,
+    isChannel: boolean,
+};
+
+export type Followable<T: CardLike> = {
+    value: T,
+    onChange: boolean => void,
+};
+
+type FollowCardProps<T: CardLike> = {
```
Comment (rahilb) [inline static/src/javascripts/projects/common/modules/identity/upsell/consent-card/FollowCard.js]: Maybe one of the flow pros from @guardian/guardian-frontend-team could see if this is a reasonable thing to do in JavaScript land?

**Comment by jfsoul** [comment]
If we aren't using "follow" as the final copy for these buttons/cards - should we be using that naming convention in the code?

**Comment by walaura** [comment]
@jfsoul i personally like instant-subscribe?

File: static/src/javascripts/projects/common/modules/identity/upsell/consent-card/FollowCard.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                </h1>
+                <p className="identity-upsell-consent-card__description">
+                    {this.props.followable.value.description}
+                </p>
+                <FollowButtonWrap
+                    following={hasFollowed}
+                    onFollow={() => {
+                        this.props.followable.onChange(
+                            true,
+                            this.props.followable.value
```
Comment (rahilb) [inline static/src/javascripts/projects/common/modules/identity/upsell/consent-card/FollowCard.js]: this smells a bit weird to me, but I can't think of an alternative that gives as nice an API to users of this component.

**Comment by rahilb** [comment]
>should we be using that naming convention in the code? 

Not too sure, I guess we can rename stuff for consistency later once the copy is all signed off?

**Review by Calum-Campbell — APPROVED** [review]
AFAICT looks good. 👍

**Pull request merged**